### ISSUE TITLE
Bar Plots and Negative Values

### ISSUE THREAD

**Issue by Benny121221** [swharden/ScottPlot#461]
**Describe the bug**
Bar Plots do not play nice with negative values:
- Unless panned/zoomed, the plot will only show y values >0 (Quadrant 1, assuming non-negative x-values)
![image](https://user-images.githubusercontent.com/8635304/84607723-2cd3cb00-ae6c-11ea-877e-589dd11c1871.png)
- Error bars are drawn from the topmost point, which for negative values is 0
![image](https://user-images.githubusercontent.com/8635304/84607734-35c49c80-ae6c-11ea-93ef-75c581dd55cf.png)

Negative values on bar plots are atypical, but I was looking at making waterfall charts:
![image](https://user-images.githubusercontent.com/8635304/84607809-9227bc00-ae6c-11ea-8ab9-ba03c6df4f8f.png)

You might recognize waterfall charts as a generalization of a Gantt chart:
![image](https://user-images.githubusercontent.com/8635304/84607827-ae2b5d80-ae6c-11ea-9504-eff3e8d012b1.png)

I will probably work on this sometime soon-ish. I noticed it because I was thinking about implementing waterfall charts as bar charts with an extra `std::shared_ptr<double> offsets` parameter, but the handling of negative values put a damper on things.

**To Reproduce**
Create a bar plot with negative y values.
```cs
//xs, ys, yError
var barPlot = plt.PlotBar(new double[]{1,2,3}, new double[]{1,0,-1}, new double[]{1,1,1});
```

**Screenshots**
I embedded them in the list :/

**Comment by Benny121221** [comment]
> ... with an extra `std::shared_ptr<double>` offsets parameter ...

Yeesh, I've been doing too much C++, I meant `double[] offsets`

**Comment by Benny121221** [comment]
Also note, negative x values cause no issues

**Comment by swharden** [comment]
Fixed in #462, thanks @Benny121221!
